src/preload.py

```diff
@@ -65,6 +65,52 @@
 PARTITIONS_CACHE = {}
 
 
+class RetryCounter:
+    """Counter and logger for the number of times that a function is retried.
+    Usage:
+        retry_counter = RetryCounter()
+        hint = "Try reducing the system entropy"
+        wrapped_func, wrap_key = retry_counter.wrap(my_func, hint, *args)
+        retry_call(wrapped_func, fargs=args, ...)
+        retry_counter.clear(wrap_key)
+    """
+
+    def __init__(self):
+        self.counter = {}
+
+    @staticmethod
+    def key(func_name, *args):
+        return " ".join(str(e) for e in (func_name,) + args)
+
+    def clear(self, key, *args):
+        del self.counter[key]
+
+    def inc(self, key, *args):
```

Comment:
I've approved a bit too fast.
`*args` are not used here.

Reply:
nor in `clear`

Reply:
Indeed. Harmless though (tested). I'll be testing further during integration with the CLI, and then I'll remove the unused args in another PR. (Waiting in case additional changes are also needed.)